hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -331,27 +334,28 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
         LOG.trace("{} not needed", c.getClass().getSimpleName());
         continue;
       }
-      sumMultiplier += multiplier;
-      total += c.cost() * multiplier;
+      if (cost < minCostNeedBalance)
+      {
+        LOG.debug("Imbalance of {} on the scale of [0, 1] is {} < threshold ({}).",
+          c.getClass().getSimpleName(), cost, minCostNeedBalance);
+      }
+      total += cost * multiplier;
     }
 
     boolean balanced = total <= 0 || sumMultiplier <= 0 ||
         (sumMultiplier > 0 && (total / sumMultiplier) < minCostNeedBalance);
     if (balanced) {
       final double calculatedTotal = total;
-      final double calculatedMultiplier = sumMultiplier;
-      sendRejectionReasonToRingBuffer(() -> getBalanceReason(calculatedTotal, calculatedMultiplier),
+      sendRejectionReasonToRingBuffer(() -> getBalanceReason(calculatedTotal),
         costFunctions);
     }
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("{} {}; total cost={}, sum multiplier={}; cost/multiplier to need a balance is {}",
-          balanced ? "Skipping load balancing because balanced" : "We need to load balance",
-          isByTable ? String.format("table (%s)", tableName) : "cluster",
-          total, sumMultiplier, minCostNeedBalance);
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("Balance decision detailed function costs={}", functionCost());
-      }
-    }
+    LOG.info("{} {} ", isByTable ? String.format("table (%s)", tableName) : "cluster",
+      balanced ? "Skipping load balancing because weighted average imbalance= "
+        + total / sumMultiplier + "< threshold (" + minCostNeedBalance + ") on the scale of [0, 1]."
+        + "If you want more aggressive balancing, either lower threshold minCostNeedbalance ("
+        + minCostNeedBalance + ") or increase the relative weight(s) of the specific cost function(s)."
+        : "Calculating plan. May take up to " + maxRunningTime + " ms to complete.");
+
```

Comment:
oof. That's a lot of logic packed into string concatenation. Can we unwrap this a bit so that future folks don't have to decipher so much?

Something like 

```
if (balanced) {
  LOG.info("{} - skipping load balancing because weighted average imbalance = {} > threshold ({})....",
      isByTable ? "Table specific ("+tableName+")" : "Cluster wide", total/sumMultiplier, minCostNeedBalance,...);
} else {
  LOG.info("{} - Calculating plan. may take up to {}ms to complete.",
      isByTable ? "Table specific ("+tableName+")" : "Cluster wide", maxRunningTime);
}
```

could be rolled into the if block that calls `sendRejectionReasonToRingBuffer`?